### PR TITLE
Switch to LLVM ORC v2

### PR DESCRIPTION
**Pull request by vtjnash**
Mostly pretty mechanical. Some old hacks went away. Result seems to work, so is presumably good?

### CODE REVIEW

**Comment by vchuravy** [comment]
And OrcV1 is gone: http://lists.llvm.org/pipermail/llvm-dev/2020-October/145866.html

**Comment by vtjnash** [comment]
remaining tasks are:
 - [ ] switch back to eager finalization (our cgmemmgr doesn't support interleaving of finalization), to address macos64 build failure
 - [ ] figure out symbol conflict from exporting function aliases
 - [ ] figure out why win32 and win64 aren't respecting the `-elf` triple

**Comment by vchuravy** [comment]
> figure out why win32 and win64 aren't respecting the -elf triple

In LLVM.jl I had to normalize the triple before passing it to target https://github.com/maleadt/LLVM.jl/blob/b0780fa710219b1ca2e0a418d5dc82fa6dfafe71/src/orc.jl#L130-L134

**Comment by vtjnash** [comment]
I suppose, but we don't usually like the default triple, and the process triple that we use appears to already call normalize: https://llvm.org/doxygen/Host_8cpp_source.html#l01545

**Review by vchuravy — APPROVED** [review]
Looks good to me! Tried this on PPC as well.

Let's squash and GTM?

File: src/debuginfo.cpp [hunk 1]
```diff
@@ -316,16 +313,15 @@ class JuliaJITEventListener: public JITEventListener
...
-        uint64_t SectionAddrCheck = 0; // assert that all of the Sections are at the same location
+        uint64_t SectionAddrCheck = 0;
+        uint64_t SectionLoadCheck = 0;
+        uint64_t SectionWriteCheck = 0;
         uint8_t *UnwindData = NULL;
 #if defined(_CPU_X86_64_)
-        uint64_t SectionLoadOffset = 1; // The real offset shouldn't be 1.
         uint8_t *catchjmp = NULL;
         for (const object::SymbolRef &sym_iter : debugObj.symbols()) {
             StringRef sName;
```
Comment (vchuravy) [inline src/debuginfo.cpp:323]:
```suggestion
```

Comment (vchuravy) [reply src/debuginfo.cpp:323]:
```
/cygdrive/c/buildbot/worker/package_win64/build/src/debuginfo.cpp: In member function 'virtual void JuliaJITEventListener::_NotifyObjectEmitted(const llvm::object::ObjectFile&, const llvm::RuntimeDyld::LoadedObjectInfo&, llvm::RTDyldMemoryManager*)':
/cygdrive/c/buildbot/worker/package_win64/build/src/debuginfo.cpp:324:18: error: conflicting declaration 'auto sName'
  324 |             auto sName = cantFail(sym_iter.getName());
      |                  ^~~~~
/cygdrive/c/buildbot/worker/package_win64/build/src/debuginfo.cpp:323:23: note: previous declaration as 'llvm::StringRef sName'
  323 |             StringRef sName;
      |                       ^~~~~
make[1]: *** [Makefile:166: debuginfo.o] Error 1
make[1]: *** Waiting for unfinished jobs....
```

File: src/jitlayers.h [hunk 2]
```diff
@@ -142,58 +139,59 @@ typedef JITSymbol JL_JITSymbol;
 // is expected.
 typedef JITSymbol JL_SymbolInfo;
 
+#if JL_LLVM_VERSION < 12000
```
Comment (vchuravy) [inline src/jitlayers.h:142]:
```suggestion
#if JL_LLVM_VERSION < 120000
```

**Pull request merged**